### PR TITLE
Updating run-gen-swagger-docs to generate docs for extension resources

### PR DESCRIPTION
**Pull request by nikhiljindal**
Ref https://github.com/kubernetes/kubernetes/issues/13883

cc @caesarxuchao @bgrant0607

### CODE REVIEW

**Comment by nikhiljindal** [comment]
You can view the generated html files using:
https://htmlpreview.github.io/?https://raw.githubusercontent.com/nikhiljindal/kubernetes/155f8383a2e5f644129d467510fcb033f2c34363/docs/api-reference/definitions.html
https://htmlpreview.github.io/?https://raw.githubusercontent.com/nikhiljindal/kubernetes/155f8383a2e5f644129d467510fcb033f2c34363/docs/api-reference/operations.html

**Comment by nikhiljindal** [comment]
This is the quickest way to get documentation but all resource definitions (for both v1 and extensions/v1beta1) are in the same file.
If we want then we can spend more time on changing gen-swagger-docs code and generate separate files, but I wasnt sure how critical that is.

@caesarxuchao Maybe you can estimate what it will take to generate separate files since you wrote the current code.

**Comment by bgrant0607** [comment]
LGTM

**Comment by caesarxuchao** [comment]
We lost the top level objects menu in definitions.html. I removed the lgtm. @bgrant0607 please add it back if you think that's not important.

**Comment by nikhiljindal** [comment]
@caesarxuchao I just ran the script. Is that not enough?
What else do I need to do?

File: hack/gen-swagger-doc/run-gen-swagger-docs.sh [hunk 1]
```diff
@@ -14,10 +14,9 @@
...
-    exit 
-fi
-OUTPUT=${2:-${PWD}}
+# Usage: run-gen-swagger-docs.sh <absolute output path, default to PWD>
 
-docker run -v ${OUTPUT}:/output gcr.io/google_containers/gen-swagger-docs:v1.1 https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/swagger-spec/$1.json https://raw.githubusercontent.com/kubernetes/kubernetes/master/pkg/api/$1/register.go
+OUTPUT=${1:-${PWD}}
+
+docker run -v ${OUTPUT}:/output gcr.io/google_containers/gen-swagger-docs:v1.1 https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/swagger-spec/v1.json https://raw.githubusercontent.com/kubernetes/kubernetes/master/pkg/api/v1/register.go
+docker run -v ${OUTPUT}:/output gcr.io/google_containers/gen-swagger-docs:v1.1 https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/swagger-spec/v1beta1.json https://raw.githubusercontent.com/kubernetes/kubernetes/master/pkg/apis/extensions/v1beta1/register.go
```
Comment (caesarxuchao) [inline hack/gen-swagger-doc/run-gen-swagger-docs.sh]: @nikhiljindal it looks like the output shouldn't be merged to one file if you specify different directory.

**Comment by caesarxuchao** [comment]
The menu should be there automatically. Could you try different directories for extensions and v1?

**Comment by caesarxuchao** [comment]
@nikhiljindal hyperlinks in operations.html are not working

**Comment by nikhiljindal** [comment]
After discussion with @caesarxuchao:
* Menu generation is hardcoded for v1. If we generate separate files for v1 and v1beta1 resources (which is easy), we will get the menu in v1 file, but not in v1beta1.
* But if we generate separate files for v1 and v1beta1, then links to v1 resources in the v1beta1 file will stop working.
So if we keep same file, then we loose the menu (index).
If we generate separate files, then links to v1 resources in v1beta1 file wont work.

Re: links in operations.html not working: They are also hardcoded for v1. They work fine for v1, but are broken for v1beta1.

@caesarxuchao is going to try to fix the code tomorrow.

**Comment by nikhiljindal** [comment]
Talked to @bgrant0607.
We should split the files and fix the bugs.
To summarize here are the issues:
* Generate index for v1beta1 files.
* Fix links to v1 definitions in v1beta1 definitions files.
* Fix links to v1 definitions in v1beta1 operations file.

**Comment by nikhiljindal** [comment]
Superseded by https://github.com/kubernetes/kubernetes/pull/15788.
Closing this one.

**Pull request closed without merging**